### DISCUSSION TITLE
Size limit in TLS session establishment

### DISCUSSION THREAD

**Discussion by JaimeGomezGarcia** [pkic/community#28] [Post-Quantum Cryptography] [answered]
Mike Ounswirth mentioned there is a size limit in the Server Hello in TLS due to TCP congestion window.
Sebastian from Bosch didn't seem to experience that issue.
Could it be clarified what the limitation is and why Bosch didn't hit it?
I don't see why fragmentation at TCP level should limit the ability for the upper layer to manage larger data.

**Comment by ounsworth** [comment, marked as answer]
Great question. 
I don't claim to be an expert here, but I will give it a try based on discussions that I've overheard.

TLS "works" with larger keys and packet fragmentation. If you're testing on a local, unsaturated, network you might not even notice the difference. But people like CloudFlare seem to care, and care _a lot_ about violating the TCP congestion window, for example see [Cloudflare blog: Sizing Up Post-Quantum Signatures](https://blog.cloudflare.com/sizing-up-post-quantum-signatures/) -- the discussion on TCP Congestion Window starts about halfway down the page. Their conclusion seems to be that congestion window violation alone can more-than-triple TLS handshake times from 100 ms to 300 ms.

**Reply by JaimeGomezGarcia** [reply]
Thank you Mike.
I'll read the post. I understand then that it may not be a TLS issue, but rather a generic TCP issue.
Anyway, now I have a clue where to search for.
